### DISCUSSION TITLE
How to export my document to a Microsoft word doc?

### DISCUSSION THREAD

**Discussion by mendax1234** [typst/typst#2561] [Q&A] [answered]
Hi, l am curious about whether l can export my document on Typst to a Microsoft Word Docs. The only option about export l see is export to a pdf. 
Thanks for any help!

**Comment by Enivex** [comment]
You cannot, but you can use pandoc to convert it. It'll be imperfect though.

**Comment by mendax1234** [comment]
> You cannot, but you can use pandoc to convert it. It'll be imperfect though.

Thanks. That maybe inconvenient. However, I am still wondering whether Typst doesn't support 'export to Microsoft word docs' function. :joy:

**Comment by frozolotl** [comment, marked as answer]
Typst itself does not support exporting documents as MS Word `.doc`/`.docx` files. That format is not well suited for sharing documents and it is difficult to export already fully formatted, laid out and typeset documents to it. It is a format meant for editing, after all.

Pandoc, however, does include a feature to convert Typst documents to Word `.docx` (not `.doc` though). Beware that styling and certain text objects will display differently or get lost entirely after that export. It is a very much imperfect conversion.
See [Installing pandoc](https://pandoc.org/installing.html) for information on installing it. The command to convert a document is simply `pandoc main.typ -o main.docx`.

**Reply by mattfbacon** [reply]
Hm, I wonder how this is compared to typst -> pdf -> docx.

**Reply by laurmaedje** [reply]
Probably pandoc produces a cleaner Word file w.r.t to the content, while PDF -> Docx produces a word file that is visually more like the original Typst document.

**Reply by jgarte** [reply]
`pandoc main.typ -o main.docx` didn't work for me:

```sh
pandoc resume.typ -o resume.docx
"resume.typ" (line 1, column 2):
"/home/jgart/.cache/typst/packages/preview/basic-resume/0.2.8/src/lib.typ" (line 6, column 2):
"/home/jgart/.cache/typst/packages/preview/basic-resume/0.2.8/src/./resume.typ" (line 1, column 2):
"/home/jgart/.cache/typst/packages/preview/scienceicons/0.1.0/index.typ" (line 2, column 18):
unexpected "`"
expecting "//", "/*", "none", "auto", "true", "false", "0b", "0x", "0o", digit, ".", "\"", "let", "set", "show", "if", "while", "for", "import", "include", "break", "continue", "return", "(", "$", "<", "{" or "["
error: Recipe `docx` failed on line 11 with exit code 64
```

**Reply by mattfbacon** [reply]
@jgarte please report this as a pandoc bug if you haven't already, assuming that the Typst code is valid and your pandoc is updated.

**Reply by jgarte** [reply]
Pandoc and Typst code is valid and up to date on Arch Linux. I can produce the PDF perfectly. Will report a bug as well to Pandoc but just wanted to let people know here that the above does not work as of yesterday.

**Reply by alerque** [reply]
@jgarte As of this writing the [pandoc-cli](https://archlinux.org/packages/extra/x86_64/pandoc-cli/) package in Arch Linux official repos is several versions behind (due to complications with new versions of Haskell not building *other* Arch packages). If you are using the Typst output or other things that have changed in the most recent releases you can run the [pandoc-bin](https://aur.archlinux.org/packages/pandoc-bin) package from the AUR instead to get the latest version.

**Comment by timoleistner** [comment]
In case someone is looking into this in the future, I've found some more options.
There is the [typ2docx](https://github.com/sghng/typ2docx?tab=readme-ov-file) package which would be worth a try. Another good option seems to be Adobe Acrobat or what worked best for me was just using Word to open the Typst generated PDF and save it as .docx.
